lib/block_producer.go

```diff
@@ -582,23 +583,28 @@ func (desoBlockProducer *DeSoBlockProducer) SignBlock(blockFound *MsgDeSoBlock)
 
 func (desoBlockProducer *DeSoBlockProducer) Start() {
 	// If we set up the max sync block height, we will not be running the block producer.
+	glog.V(1).Infof("DeSoBlockProducer.Start() called. MaxSyncBlockHeight: %v",
+		desoBlockProducer.chain.MaxSyncBlockHeight)
 	if desoBlockProducer.chain.MaxSyncBlockHeight > 0 {
 		glog.V(2).Infof("DeSoBlockProducer.Start() exiting because "+
 			"MaxSyncBlockHeight: %v is greater than 0", desoBlockProducer.chain.MaxSyncBlockHeight)
 		return
 	}
 
 	// Set the time to a nil value so we run on the first iteration of the loop.
-	var lastBlockUpdate time.Time
+	lastBlockUpdate := time.Now()
 	desoBlockProducer.producerWaitGroup.Add(1)
 
 	for {
-		if atomic.LoadInt32(&desoBlockProducer.exit) >= 0 {
+		if atomic.LoadInt32(&desoBlockProducer.exit) > 0 {
```

Comment:
This was causing a lot of the erratic behavior when nodes would restart. See main comment.